R/bind_epiparam.R

```diff
@@ -0,0 +1,119 @@
+#' of `<epidist>` objects. It can also be a data frame as long as the columns
+#' conform to the columns of an `<epiparam>` object.
+#'
+#' @return An `<epiparam>` object
+#' @export
+#'
+#' @examples
+#' eparam <- epiparam()
+#' edist <- epidist(
+#'   disease = "ebola",
+#'   epi_dist = "incubation_period",
+#'   prob_distribution = "lnorm",
+#'   prob_distribution_params = c(meanlog = 1, sdlog = 1)
+#' )
+#' bind_epiparam(eparam, edist)
+bind_epiparam <- function(epiparam, epi_obj) {
+
+  # check input
+  validate_epiparam(epiparam)
+  if (is_epidist(epi_obj)) {
+    validate_epidist(epi_obj)
+  } else if (inherits(epi_obj, "list")) {
+    lapply(epi_obj, validate_epidist)
+  } else if (is_vb_epidist(epi_obj)) {
+    validate_vb_epidist(epi_obj)
+  } else if (is_epiparam(epi_obj)) {
+    validate_epiparam(epi_obj)
+  } else if (is.data.frame(epi_obj)) {
+    stopifnot(
+      "<data.frame> given must have the same column names as <epiparam>" =
+        setequal(colnames(epi_obj), colnames(epiparam))
+    )
+  } else {
+    stop(
+      "Only <epidist>, <vb_epidist> or <epiparam> can bind to <epiparam>",
+      call. = FALSE
+    )
+  }
+
+  # convert epidist to epiparam object
+  if (is_epidist(epi_obj) || inherits(epi_obj, "list") ||
+      is_vb_epidist(epi_obj)) {
+    epi_obj <- as_epiparam(x = epi_obj)
+  }
+
+  # keep matching columns between objects
+  epiparam <- epiparam[, intersect(colnames(epiparam), colnames(epi_obj))]
+
+  # bind epidist to epi
+  out <- rbind(epiparam, epi_obj)
+
+  # validate new epiparam object
+  validate_epiparam(out)
+
+  # return new epiparam object
+  out
+}
+
+#' Bind columns to an `<epiparam>` object
+#'
+#' @description Adds columns to an existing `<epiparam>` object.
+#'
+#' This function avoids the unclassing feature of [`cbind()`]. The work around
+#' by using `[<-` instead of `cbind()` was adapted from
+#' [`tibble::add_column()`].
+#'
+#' @details Note that if the variables passed to `cbind_epiparam()` are not of
+#' equal length to the number of rows in the `<epiparam>` object then R's
+#' recycling rules apply, see
+#' <https://r4ds.had.co.nz/vectors.html#scalars-and-recycling-rules>
+#'
+#' Adding columns should not invalidate the `<epiparam>` object, but if it does
+#' the function will throw an error and not return an `<epiparam>` object.
+#'
+#' @inheritParams bind_epiparam
+#' @param ... Variables to be added as columns to the `<epiparam>` object.
+#'
+#' @return An `<epiparam>` object
+#' @export
+#'
+#' @examples
+#' eparam <- epiparam()
+#' cbind_epiparam(eparam, extra_col = NA)
+cbind_epiparam <- function(epiparam, ...) {
+
+  # capture ... as data frame
+  df <- data.frame(...)
+
+  # solution from https://github.com/tidyverse/tibble/blob/main/R/add.R
+  end_pos <- ncol(epiparam) + seq_len(ncol(df))
+  epiparam[end_pos] <- df
+
+  # validate new epiparam object
+  validate_epiparam(epiparam)
+
+  # return epiparam
+  epiparam
+}
```

Comment:
I see your point. This could be a principle of the package that we could document in a design vignette. For example, the `<epiparam>` is class is thin wrapper around `<data.frame>`, for all data manipulation use `{dplyr}`.

I think the best way to proceed to to keep the namespace similar to now for the release and have the next release target be to minimise parts of the namespace by merging or removing what we feel is unimportant.

Reply:
Resolved in #171.